components/SideBar.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { SVGProps } from 'react';
+import { useGlobalContext } from './context';
+import Link from 'next/link';
+import {
+  LogoutIcon,
+  EditIcon,
+  LocationIcon,
+  DashboardIcon,
+  ExitIcon,
+} from './icons/sidebar_icons';
+
+const Sidebar = () => {
+  const { isSidebarOpen, closeSidebar } = useGlobalContext();
+
+  return (
+    <div
+      className={`transition-all  duration-500  fixed top-0 ${
+        isSidebarOpen ? 'left-0' : '-left-64'
+      }`}
+    >
+      <div className="flex h-screen overflow-y-auto flex-col bg-ocf-black w-64 px-4 py-8 min-h-screen relative">
+        <button
+          onClick={closeSidebar}
+          className="absolute top-1 right-1 text-white w-8 h-8 rounded-full flex items-center justify-center focus:outline-none ml-6"
+        >
+          <ExitIcon />
+        </button>
+        <div className="text-xs	flex flex-col mt-6 justify-between flex-1">
+          <Button
+            href="/dashboard"
+            name="Dashboard"
+            svg={<DashboardIcon />}
+            className="mx-4 font-medium flex-1 align-center text-amber"
+          />
+          <div className="text-xs">
+            <Button
+              href="/form"
+              name="Add a Location"
+              svg={<LocationIcon />}
+              className="mx-4 font-medium flex-1 align-center text-white"
+            ></Button>
+            <Button
+              href="/form"
+              name="Edit Site Details"
+              svg={<EditIcon />}
+              className="mx-4 font-medium flex-1 align-center text-white"
+            ></Button>
+            <Button
+              href="/api/auth/logout"
+              name="Logout"
+              svg={<LogoutIcon />}
+              className="mx-4 font-medium flex-1 align-center text-white"
+            ></Button>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+type SideBarButtonProps = {
+  href: string;
```

Comment:
Instead of specifying `href` here, a more extensible solution would be to import `LinkProps` from `next/link`, and then replace `href: string;` with `linkProps: LinkProps`. Then, instead of `<Link href={href}>` we can actually do 
`<Link {...linkProps}>` and allow more than just the `href` prop to potentially be passed in!